### PR TITLE
Add Path module to Node bindings, documented

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
@fable-compiler/code-maintenance 
@fable-compiler/documentation Please review :pray:

Not exposing the types just yet until we figure out the best way to do so.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
Great, including the comments would be incredibly helpful for users! :clap:

**Comment by jmmk** [comment]
@Zaid-Ajaj 
> Not exposing the types just yet until we figure out the best way to do so.

What's blocking? Finalization of https://github.com/fable-compiler/Fable/pull/929?

**Comment by Zaid-Ajaj** [comment]
@jmmk Yes, I am not sure what the eventual consensus was on how to do this. There is also [another discussion](https://github.com/fable-compiler/Fable/issues/930) on putting Fable.Import the bindings in different repo's or libraries so I am just waiting here :)

File: src/dotnet/Fable.Core/Import/Node/Path.fs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+[<AutoOpen>]
+module path_types = 
+    type PathObjectProps = 
+        abstract dir : string
+        abstract root : string
+        [<Emit("$0.base{{=$1}}")>] // using this because back ticks are way too ugly to read in code
+        abstract baseName : string
+        abstract name : string
+        abstract ext : string
```
Comment (alfonsogarciacaro) [inline src/dotnet/Fable.Core/Import/Node/Path.fs]: If we want to use `createEmpty<PathObjectProps>` with this, we will have to make all the fields mutable, that is adding ` with get, set` and the end of every field.

**Comment by alfonsogarciacaro** [comment]
@Zaid-Ajaj Ok, let's assume the consensus has been reached with #929 😄 So:

- No nested `_types` module
- The `Globals` type is renamed to `IExports`
- We'll have a value in Fable.Import.Node.Exports.fs to represent `Path.IExports`.

Let's merge all the pending PRs here and then I'll create a new repo for Fable.Import.Node so we can continue work there :+1:

File: src/dotnet/Fable.Core/Import/Node/Path.fs (hunk 1 above)
Comment (Zaid-Ajaj) [inline src/dotnet/Fable.Core/Import/Node/Path.fs]: Wow, I totally forget 😅 fixed in latest commit

**Comment by Zaid-Ajaj** [comment]
@alfonsogarciacaro That's great news! I corrected the code as suggested except the last point becuase if I don't have the file Fable.Import.Node.Exports.fs on my (now out-dated) branch so I will fix it in another PR

**Comment by jmmk** [comment]
@Zaid-Ajaj You may already know this, but I'll describe below the process for rebasing your work onto `master` or merging the latest `master` into your branch

Assuming you have your fork  (`Zaid-Ajaj/Fable`) cloned locally and set as `origin`:

- Add the fable repo as a remote: `git remote add fable-origin git@github.com:fable-compiler/Fable.git`
- Pull the latest master code: `git checkout master && git pull fable-origin master`
- Switch to your branch: `git checkout node/path-module`

And do one of the following:

1. `git merge master`
2. `git rebase master`  - The rebase may require you to `force` push, so I would only recommend this option if you understand it

This will get your feature branch (`node/path-module`) up to date with the latest `master` changes, and you should then be able to add `Path.IExports` to the `Exports` file

**Comment by Zaid-Ajaj** [comment]
@jmmk Thanks a lot! I still don't understand why that worked but I am happy it did :smile: I will print-screen your comment for future reference :wink:

**Comment by jmmk** [comment]
@Zaid-Ajaj 

For the `git remote add` and `git pull` part:

A repository can point to any number of remotes. This is just the address of a git repository. So we added the Fable repository as a new remote and updated your local `master` branch with all the changes from that remote's `master` branch (known as a "fast forward").

---

For the merge:

`master` and `node/path-module` had a common ancestor commit in the past (whatever was on `master` when you created your branch). When you merge the latest `master`, it looks at every change made by both `master` and your branch _since that common commit_ and attempts to reconcile the changes. 

It will attempt to add every single change from both branches (new files, deleted files, modified files). If both branches have modified the same files, there can be conflicts (things git does not know how to merge automatically).

**Comment by inosik** [comment]
Unfortunately, the new file broke the build. Fix in #938.

**Comment by Zaid-Ajaj** [comment]
@jmmk That makes sense, because I only worked on my file and didn't change what others have been working on, I didn't get conflicts

Thanks for the explaination, that makes two screen shots :wink:

**Pull request merged**